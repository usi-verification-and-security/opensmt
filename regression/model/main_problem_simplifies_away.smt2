(set-option :produce-models true)
(set-logic QF_UF)

(set-info :status sat)
(declare-sort U 0)
(declare-fun Add (U) U)
(declare-fun b () Bool)
(declare-fun u () U)
(declare-fun t () U)
(declare-fun z () U)

(assert (= t (Add u)))
(assert (= b (not (= u z))))
(assert (not b))

(check-sat)
(get-model)
(exit)
