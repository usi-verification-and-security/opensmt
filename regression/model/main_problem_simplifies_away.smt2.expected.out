sat
(
  (define-fun Add ((x0 U)) U
    (ite (= (as @2 U) x0) (as @3 U) (as @d2 U)))
  (define-fun b () Bool
    false)
  (define-fun u () U
    (as @2 U))
  (define-fun t () U
    (as @3 U))
  (define-fun z () U
    (as @2 U))
)
